### PR TITLE
Add com.krioltech.fnmap

### PR DESCRIPTION
**Pull request by gmckesey**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
com.krioltech.fnmap is a frontend to the nmap security scanner written in flutter
- [X ] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ X] My pull request follows the instructions at [App Submission][submission].
- [ X] I have [built][build] and tested the submission locally.
- [X ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
_home_ permissions are being requested in order to save scans and load scans.
- [ X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X ] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [ X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: flathub.json
Comment (hfiguiere) [inline flathub.json:1]: this file is invalid.

File: com.krioltech.fnmap.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - mkdir -p /app/share/icons
+      - ln -s /app/fnmap/fnmap /app/bin/fnmap
+      - install -Dm644 fnmap-archive/com.krioltech.fnmap.metainfo.xml /app/share/appdata/com.krioltech.fnmap.appdata.xml
+      - install -Dm644 fnmap-archive/fnmap.desktop /app/share/applications/fnmap.desktop
+      - install -Dm644 fnmap-archive/icons/fnmap.png /app/share/icons/com.krioltech.fnmap.png
+      - install -Dm644 fnmap-archive/icons/nmap.png /app/share/icons/com.krioltech.nmap.png
+    sources:
+      - type: archive
+#        path: ../../fnmap-flatpak/archives/fnmap-1.2.1.x64.tar.gz
+        url: https://github.com/gmckesey/fnmap-flatpak/raw/main/archives/fnmap-1.2.1.x64.tar.gz
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:43]: you should NEVER checkout from a git branch  (`main` in that case)

File: com.krioltech.fnmap.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - cp -R fnmap-archive/bin/bundle /app/fnmap
+      - chmod +x /app/fnmap/fnmap
+      - mkdir -p /app/bin/
+      - mkdir -p /app/lib/
+      - mkdir -p /app/share/icons
+      - ln -s /app/fnmap/fnmap /app/bin/fnmap
+      - install -Dm644 fnmap-archive/com.krioltech.fnmap.metainfo.xml /app/share/appdata/com.krioltech.fnmap.appdata.xml
+      - install -Dm644 fnmap-archive/fnmap.desktop /app/share/applications/fnmap.desktop
+      - install -Dm644 fnmap-archive/icons/fnmap.png /app/share/icons/com.krioltech.fnmap.png
+      - install -Dm644 fnmap-archive/icons/nmap.png /app/share/icons/com.krioltech.nmap.png
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:39]: 1. this line is duplicated
2. this is an invalid location for the icon.

File: com.krioltech.fnmap.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        url: https://nmap.org/dist/nmap-7.94.tgz
+        sha256: 7ebc3a0d369d5965ba8b6513bce3ff6d3307a7cc87dd18f70d0af5d8e66a2849
+  - buildsystem: simple
+    name: fnmap
+    build-commands:
+      - cp -R fnmap-archive/bin/bundle /app/fnmap
+      - chmod +x /app/fnmap/fnmap
+      - mkdir -p /app/bin/
+      - mkdir -p /app/lib/
+      - mkdir -p /app/share/icons
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:34]:
```suggestion
```

`install -D` does that for you.

File: com.krioltech.fnmap.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+app-id: com.krioltech.fnmap
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:1]: is this controlled by the upstream developers?

I get `404 Not Found`

File: com.krioltech.fnmap.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+command: fnmap
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --filesystem=home
+  - --filesystem=xdg-documents
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:14]: why do you need home access for `nmap`? Also these two filesystem permission are redundant.

File: com.krioltech.fnmap.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      - type: archive
+        url: https://nmap.org/dist/nmap-7.94.tgz
+        sha256: 7ebc3a0d369d5965ba8b6513bce3ff6d3307a7cc87dd18f70d0af5d8e66a2849
+  - buildsystem: simple
+    name: fnmap
+    build-commands:
+      - cp -R fnmap-archive/bin/bundle /app/fnmap
+      - chmod +x /app/fnmap/fnmap
+      - mkdir -p /app/bin/
+      - mkdir -p /app/lib/
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:33]:
```suggestion
```
this is never used it seems.

File: com.krioltech.fnmap.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      - mkdir -p /app/share/icons
+      - ln -s /app/fnmap/fnmap /app/bin/fnmap
+      - install -Dm644 fnmap-archive/com.krioltech.fnmap.metainfo.xml /app/share/appdata/com.krioltech.fnmap.appdata.xml
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:36]:
```suggestion
      - install -Dm644 fnmap-archive/com.krioltech.fnmap.metainfo.xml -t /app/share/metainfo/
```

File: com.krioltech.fnmap.yml (hunk 2 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml:39]: One of the lines is fnmap the other is nmap they are different.

File: com.krioltech.fnmap.yml (hunk 4 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml:1]: The web site is https://www.krioltech.com

File: com.krioltech.fnmap.yml (hunk 5 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml:14]: I need it for saving and loading nmap scan xml files.

File: com.krioltech.fnmap.yml (hunk 2 above)
Comment (hfiguiere) [reply com.krioltech.fnmap.yml:39]: thing is `com.krioltech.nmap.png` doesn't match the app id so I don't see how it's useful.

File: com.krioltech.fnmap.yml (hunk 4 above)
Comment (hfiguiere) [reply com.krioltech.fnmap.yml:1]: image

Comment (hfiguiere) [reply com.krioltech.fnmap.yml:1]: without www
image

File: com.krioltech.fnmap.yml (hunk 1 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: _fixed_ I am now using the v1.2.1 tag.

File: com.krioltech.fnmap.yml (hunk 2 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: I have removed this line.

File: com.krioltech.fnmap.yml (hunk 3 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: Removed.

File: com.krioltech.fnmap.yml (hunk 4 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml:1]: Should work now. https://www.krioltech.com

File: com.krioltech.fnmap.yml (hunk 6 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: removed.

File: com.krioltech.fnmap.yml (hunk 7 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: Done.

File: com.krioltech.fnmap.yml (hunk 5 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: Redundant home permission removed.

File: com.krioltech.fnmap.yml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      - cp -R fnmap-archive/bin/bundle /app/fnmap
+      - chmod +x /app/fnmap/fnmap
+      - mkdir -p /app/bin/
+      - ln -s /app/fnmap/fnmap /app/bin/fnmap
+      - install -Dm644 fnmap-archive/com.krioltech.fnmap.metainfo.xml -t /app/share/metainfo/
+      - install -Dm644 fnmap-archive/fnmap.desktop /app/share/applications/fnmap.desktop
+      - install -Dm644 fnmap-archive/icons/fnmap.png /app/share/icons/com.krioltech.fnmap.png
+    sources:
+      - type: archive
+#        path: ../../fnmap-flatpak/archives/fnmap-1.2.1.x64.tar.gz
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:38]:
```suggestion
```

File: com.krioltech.fnmap.yml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --filesystem=xdg-documents
+rename-desktop-file: fnmap.desktop
+#rename-icon: fnmap
+#copy-icon: true
+#rename-appdata-file: fnmap.appdata.xml
```
Comment (hfiguiere) [inline com.krioltech.fnmap.yml:17]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build com.krioltech.fnmap

File: com.krioltech.fnmap.yml (hunk 4 above)
Comment (bbhtt) [reply com.krioltech.fnmap.yml:1]: I'm getting a bad gateway still

Comment (gmckesey) [reply com.krioltech.fnmap.yml:1]: Ok.  Please try again.

File: com.krioltech.fnmap.yml (hunk 9 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: Accepted changes.

File: com.krioltech.fnmap.yml (hunk 8 above)
Comment (gmckesey) [reply com.krioltech.fnmap.yml]: Accepted changes.

File: com.krioltech.fnmap.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+command: fnmap
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --share=network
+  - --filesystem=home
+  - --filesystem=xdg-documents
```
Comment (bbhtt) [inline com.krioltech.fnmap.yml:14]: xdg-documents is redundant with home. Why do you need `home`?

**Review by bbhtt — CHANGES_REQUESTED** [review]
Please rebase properly to `new-pr` and remove the extra commits and the files from `master` branch.

**Pull request closed without merging**